tiledb/api/c_api/group/group_api.cc

```diff
@@ -0,0 +1,797 @@
+    char** name) {
+  ensure_group_is_valid(group);
+  ensure_output_pointer_is_valid(uri);
+  ensure_output_pointer_is_valid(type);
+  ensure_output_pointer_is_valid(name);
+
+  auto&& [st, uri_str, object_type, name_str] =
+      group->group().member_by_index(index);
+  throw_if_not_ok(st);
+
+  *uri = static_cast<char*>(std::malloc(uri_str.value().size() + 1));
+  if (*uri == nullptr) {
+    // PJD: Should this be OOM instead of ERR? Try to not rock
+    // the boat for the time being I guess? Same as for name below.
+    return TILEDB_ERR;
+  }
+  std::memcpy(*uri, uri_str.value().data(), uri_str.value().size());
+  (*uri)[uri_str.value().size()] = '\0';
+
+  *type = static_cast<tiledb_object_t>(object_type.value());
+
+  *name = nullptr;
+  if (name_str.has_value()) {
+    *name = static_cast<char*>(std::malloc(name_str.value().size() + 1));
+    if (*name == nullptr) {
+      return TILEDB_ERR;
+    }
+
+    std::memcpy(*name, name_str.value().data(), name_str.value().size());
+    (*name)[name_str.value().size()] = '\0';
+  }
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_group_get_member_by_name(
+    tiledb_group_handle_t* group,
+    const char* name,
+    char** uri,
+    tiledb_object_t* type) {
+  ensure_group_is_valid(group);
+  if (name == nullptr) {
+    throw CAPIStatusException("argument `name` may not be nullptr");
+  }
+  ensure_output_pointer_is_valid(uri);
+  ensure_output_pointer_is_valid(type);
+
+  auto&& [st, uri_str, object_type, name_str, ignored_relative] =
+      group->group().member_by_name(name);
+  throw_if_not_ok(st);
+
+  *type = static_cast<tiledb_object_t>(object_type.value());
+  *uri = static_cast<char*>(std::malloc(uri_str.value().size() + 1));
+  if (*uri == nullptr) {
+    return TILEDB_ERR;
+  }
+  std::memcpy(*uri, uri_str.value().data(), uri_str.value().size());
+  (*uri)[uri_str.value().size()] = '\0';
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_group_get_is_relative_uri_by_name(
+    tiledb_group_handle_t* group,
+    const char* name,
+    uint8_t* is_relative) {
+  ensure_group_is_valid(group);
+  if (name == nullptr) {
+    throw CAPIStatusException("argument `name` may not be nullptr");
+  }
+  ensure_output_pointer_is_valid(is_relative);
+
+  auto&& [st, uri_str, object_type, name_str, relative] =
+      group->group().member_by_name(name);
+  throw_if_not_ok(st);
+
+  *is_relative = *relative ? 1 : 0;
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_group_is_open(
+    tiledb_group_handle_t* group, int32_t* is_open) {
+  ensure_group_is_valid(group);
+  ensure_output_pointer_is_valid(is_open);
+
+  *is_open = static_cast<int32_t>(group->group().is_open());
+
+  return TILEDB_OK;
+}
+
+capi_return_t tiledb_group_get_uri(
+    tiledb_group_handle_t* group, const char** group_uri) {
+  ensure_group_is_valid(group);
+  ensure_output_pointer_is_valid(group_uri);
+
+  // PJD: Is this awkward that the other string returning
+  // functions allocated storage, but this one does not?
```

Comment:
Yes, it's groundless inconsistency. This should also be solved by having a handle type for returned-strings.